Create liftsubsystem

lbrenn7 opened the pull request:
created break commands to engage or disengage the break depending on if the switch is on. created movelift command to give the motor a speed unless it's at the upper or lower limit. created a command to move the arm if the switch is pressed. created a home position command that makes the lift height at 0. created the gotoHeight command which moves the lift to different predetermined heights using enums based on which part of the arena the robot is in and how high the lift should go up

lbrenn7 committed:
Added Life Subsystem

All command outlines were added. Different heights were put in to one command with enums created for all heights.

lbrenn7 committed:
Added move lift, Move Arm and Brake commands

Move lift command respects upper and lower limits. Brake command can be engaged and disengaged. Added variables to Robot map.

lbrenn7 committed:
go to Height Command for lift

Defined enums for height command and finished height command

lbrenn7 committed:
Home Position command for lift

finished the home position command which is just the "floor load" at 0.0

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -0,0 +1,206 @@
+                double liftSpeed;
+                liftSpeed = Robot.xBoxCoPilot.getY(Hand.kRight);
+                // Called repeatedly when this Command is scheduled to run
+                if (upperLimit.get()) {
```
You can combine these two: `if (upperLimit.get() && liftSpeed > 0.0)`, same with the one below

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -0,0 +1,206 @@
+        };
+    }
+
+    public Command moveArm() {
```
Does this do the same thing as the one above it?

msoucy replied:
This is for the arm - not a problem.

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -0,0 +1,206 @@
+            @Override
+            protected void execute() {
+                double currentHeight = heightEncoder.getDistance();
+                if (currentHeight < height.get()) {
```
Add checks for the limit switches like the one above

lbrenn7 committed:
Update gotoHeight Command and moveLift command

in the moveLift command I combined 2 blocks of if statements in to one if statement twice and I checked the height for the limit switches

msoucy committed:
Merge branch 'master' into master
